src/Compilers/Core/Portable/Syntax/SyntaxNode.cs

```diff
@@ -421,7 +421,7 @@ public bool ContainsDiagnostics
         /// </summary>
         public bool Contains(SyntaxNode node)
         {
-            if (node == null || !node.FullSpan.IntersectsWith(node.FullSpan))
+            if (node == null || !this.FullSpan.IntersectsWith(node.FullSpan))
```

Comment:
`IntersectsWith` returns true if the end point of one span matches the start of the other. Consider using ```this.FullSpan.Contains(node.FullSpan)``` instead.